client/state/sharing/keyring/actions.js

```diff
@@ -12,9 +17,11 @@ import {
 /**
  * Triggers a network request for a user's connected services.
  *
+ * @param {Function} successCallback Function that is called after a successful
+ *                                   request. Default: _.identity.
  * @return {Function} Action thunk
  */
-export function requestKeyringConnections() {
+export function requestKeyringConnections( successCallback = identity ) {
```

Comment:
Besides middleware (which doesn't seem helpful in this case) the only other way I can think of doing this is to set some state for `isSelectingAccount` (which isn't persisted) in the reducer on `KEYRING_CONNECTIONS_REQUEST_SUCCESS and clear it `KEYRING_CONNECTIONS_REQUEST` or `KEYRING_CONNECTIONS_REQUEST_FAILURE`.

Any thoughts or @dmsnell @aduth ?